__tests__/app.test.js

```diff
@@ -39,103 +39,175 @@ describe("/api", () => {
-        .then(({ body }) => {
-          expect(body.msg).toBe("No Comments Found");
+        expect(body.comments).toBeSortedBy("created_at", {
+          coerce: true,
+          descending: true,
         });
-    });
-    test("returns a 400 Bad Request when an invalid id is requested", () => {
-      return request(app)
-        .get("/api/articles/banana/comments")
-        .expect(400)
-        .then(({ body }) => {
-          expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("GET:200 works when there are no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toEqual([]);
+      });
+  });
+  test("returns a 404 Not Found when a valid but non-existing id is requested", () => {
+    return request(app)
+      .get("/api/articles/999999/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Resource Not Found");
+      });
+  });
+  test("returns a 400 Bad Request when an invalid id is requested", () => {
+    return request(app)
+      .get("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
+
+describe("POST /api/articles/:articleId/comments", () => {
+  test("POST returns a 201 and the comment successfully added", () => {
+    const comment = {
+      username: "rogersop",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(comment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toMatchObject({
+          body: "This is a comment",
+          author: "rogersop",
+          article_id: 2,
         });
-    });
-  });
\ No newline at end of file
+      });
+  });
+  test("POST returns a 400 and bad request when insufficient info provided", () => {
+    const comment = {
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+  test("POST returns a 404 and resource not found when article_id doesn't exist", () => {
+    const comment = {
+      username: "rogersop",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/9999999/comments")
+      .send(comment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Resource Not Found");
+      });
+  });
+  test("POST returns a 400 and bad request when article_id invalid", () => {
+    const comment = {
+      username: "rogersop",
+      body: "This is a comment",
+    };
+    return request(app)
+      .post("/api/articles/invalidId/comments")
+      .send(comment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad Request");
+      });
+  });
+});
```

Comment:
Really good error handling but there is a missing error handling test which is very important for this endpoint point. When given an none-existant username i.e. 
```
const comment = {
      username: "not-a-user",
      body: "This is a comment",
    };
```
This should give us back a 404 as well as this user doesnt not exist.